### PR TITLE
new recipe: gencove

### PR DESCRIPTION
**Pull request by matthdsm**
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by matthdsm** [comment]
ping @atrigila

File: recipes/gencove/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    - requests >=2.19.1
+    - boto3 >=1.17.97
+    - progressbar2 ==3.55.0
+    - backoff ==1.11.0
+    - pydantic ==1.9.2
+    - pytest-runner
+
+test:
+  commands:
+    - gencove upload -h
```
Comment (atrigila) [inline recipes/gencove/meta.yaml:35]: what about changing test to `gencove --help`?

Comment (matthdsm) [reply recipes/gencove/meta.yaml:35]:
```suggestion
    - gencove --help
```

File: recipes/gencove/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    - python >=3.7
+    - pytest-runner
+  run:
+    - python >=3.7
+    - python-dateutil >=2.2.0
+    - six >=1.5
+    - click >=7.0
+    - requests >=2.19.1
+    - boto3 >=1.17.97
+    - progressbar2 ==3.55.0
```
Comment (pinin4fjords) [inline recipes/gencove/meta.yaml:31]: Do these pins really need to be so precise? Is a range not possible?

File: recipes/gencove/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+test:
+  commands:
+    - gencove --help
+
+about:
+  home: https://docs.gencove.com
+  license: Apache-2.0
+  license_file: LICENSE
+  summary: The Gencove command-line interface (CLI) can be used to easily access the API.
```
Comment (pinin4fjords) [inline recipes/gencove/meta.yaml:44]: Could you improve this a bit for those who might not know what Gencove is?

File: recipes/gencove/meta.yaml (hunk 2 above)
Comment (matthdsm) [reply recipes/gencove/meta.yaml:31]: I'm not really sure, the requirements are copied from the repo, so I'm hesitant to change anything.
Also not my piece of software, so I've no idea

File: recipes/gencove/meta.yaml (hunk 3 above)
Comment (matthdsm) [reply recipes/gencove/meta.yaml:44]:
```suggestion
  summary: Gencove is a high-throughput, cost-effective platform for genome sequencing and analysis. The Gencove command-line interface (CLI) can be used to easily access the API.
```

Comment (matthdsm) [reply recipes/gencove/meta.yaml:44]: Like this?

File: recipes/gencove/meta.yaml (hunk 2 above)
Comment (pinin4fjords) [reply recipes/gencove/meta.yaml:31]: Exact pins like this are usually a recipe for dependency hell in the future, but I won't fight you on it

File: recipes/gencove/meta.yaml (hunk 3 above)
Comment (pinin4fjords) [reply recipes/gencove/meta.yaml:44]: Maybe 'the API' -> 'the Gencove API'? Otherwise good

Comment (matthdsm) [reply recipes/gencove/meta.yaml:44]: Fixed it!

File: recipes/gencove/meta.yaml (hunk 2 above)
Comment (matthdsm) [reply recipes/gencove/meta.yaml:31]: I wholly agree, but I also don't want to break stuff

**Comment by matthdsm** [comment]
@pinin4fjords, can I get an approve?

**Review by pinin4fjords — APPROVED** [review]
Looks good!

**Comment by matthdsm** [comment]
Thanks @pinin4fjords

**Comment by rpetit3** [comment]
@matthdsm https://quay.io/repository/biocontainers/gencove is now available

**Comment by matthdsm** [comment]
Thanks 🙏

**Pull request merged**